### DISCUSSION TITLE
Preflight requests to raw.githubusercontent.com

### DISCUSSION THREAD

**Discussion by kronoslang** [community/community#24659] [Apps, API and Webhooks]
<p>Hi,</p>
<p>I’m trying to pull raw files from GitHub via js in browser. Public files work great, but when I try to get private files, the request fails due to <a href="http://raw.githubusercontent.com" rel="nofollow noopener">raw.githubusercontent.com</a> *rejecting the OPTIONS preflight request* with 403.</p>
<p>Curling on command line, it works fine with an access token, because curl doesn’t preflight.</p>
<pre><code class="lang-auto">
\&gt;curl -is https://$ACCESS\_TOKEN@raw.githubusercontent.com/$SLUG/master/module.json  
HTTP/1.1 200 OK

</code></pre>
<p>If I manually preflight, the problem appears again:</p>
<pre><code class="lang-auto">\&gt; curl -is -H "Origin: http://localhost:3499" -H "Access-Control-Request-Method:GET" -H "Access-Control-Request-Headers: X-Requested-With" -X OPTIONS https://$ACCESS\_TOKEN@raw.githubusercontent.com/$SLUG/master/module.json  
HTTP/1.1 403 Forbidden  
</code></pre>
<p>I can get the file contents from the content API, but that’s clunky because stuff always seems to be base64, even with the raw media type (seems to just strip enclosing json metadata).</p>
<p>Could you please consider looking into supporting OPTIONS on <a href="http://raw.githubusercontent.com" rel="nofollow noopener">raw.githubusercontent.com</a>? Or is there something I could do with the content api to get raw file contents?</p>
<p>Thanks!</p>
<p>Vesa</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Thanks for reaching out.</p>
<p>I’ll pass along the feedback to the development team. I can’t make any promises or give an ETA on when something might change, but your request will be in the hands of the appropriate people.</p>
<p>Thanks again for letting us know.</p>

**Reply by samarth-math** [reply]
I'm seeing the same issues on a basic jekyll pages site
`curl -is -H "Origin: http://google.com" -H "Access-Control-Request-Method:GET" -H "Access-Control-Request-Headers: X-Requested-With" -X OPTIONS blog.anamii.com/latest-posts` // (main domain same as anamii-org.github.io)

The browser preflight fails, but the original request succeeds.

the pre-flight fails but the regular curl fetch request succeeeds.
It seems like this would be a straightforward nginx fix to handle OPTIONS requests correctly.

**Comment by avlemos** [comment]
<p>I am getting the same problem but with simple public files. Was there a recent change?</p>

**Comment by satyaraihpe09** [comment]
<p>I am also facing the same problem. Is there any work around for the same or is there any changes done for it?</p>

**Comment by starpit** [comment]
<p>what’s the story here? <a href="http://raw.githubusercontent.com" rel="noopener nofollow ugc">raw.githubusercontent.com</a> still 403’s all CORS Preflight requests, even from content served by github pages.</p>

**Comment by starpit** [comment]
<p>Howdy! What is the status of this issue?</p>

**Comment by darwinjob** [comment]
It seems the issue is still here

**Comment by commandhat** [comment]
Issue still present in 2023. @lee-dohm Can you follow up on this? Manual pre-flights still get an automatic 403.

**Comment by lucasgelfond** [comment]
Still here in 2024, LOL!

**Comment by ZubaeyrMSFT** [comment]
Same here in 2024!! Things like this should be basics for such a big org.

**Comment by dhjw** [comment]
So much for my idea of doing If-None-Match to get a Not Modified response on a slow-growing data file and saving GitHub and my users' bandwidth.
